### DISCUSSION TITLE
when should we execute `stop` of `useIntersectionObserver`, on unMount?

### DISCUSSION THREAD

**Discussion by mesqueeb** [vueuse/vueuse#2307] [Q&A] [answered]
when should we execute `stop` of `useIntersectionObserver`, on `unMount`?

Or is this done for us?

**Comment by darrylnoakes** [comment, marked as answer]
It is done automatically.

This is the relevant line in the source: [VueUse GitHub - useIntersectionObserver - Line 86](https://github.com/vueuse/vueuse/blob/main/packages/core/useIntersectionObserver/index.ts#L86).

The returned `stop` function is for if you want to stop the observer at some time other than unmount.
